internal/services/customization/interface.go

```diff
@@ -0,0 +1,36 @@
+package customization
+
+import (
+	"context"
+
+	"github.com/Azure/terraform-provider-azapi/internal/clients"
+	"github.com/Azure/terraform-provider-azapi/internal/services/parse"
+)
+
+type Resource interface {
+	GetResourceType() string
+	CreateFunc() Func
+	UpdateFunc() Func
+	DeleteFunc() DeleteFunc
+	ReadFunc() Func
+}
+
+type Func func() error
+
+type DeleteFunc func(ctx context.Context, clients clients.Client, id parse.ResourceId, options clients.RequestOptions) error
+
+var customizations = make(map[string]Resource)
+
+func init() {
+	var keyVaultKeyCustomization Resource = KeyVaultKeyCustomization{}
+	customizations[keyVaultKeyCustomization.GetResourceType()] = keyVaultKeyCustomization
+
+}
+
+func GetCustomization(resourceType string) *Resource {
```

Comment:
Given that the interface will store a pointer to the concrete type, consider making this return a `Resource` rather than a `*Resource`.

Also consider returning a bool value, making it similar to the idiomatic go: `if val, ok := GetCustomization(s); ok {}`